hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationPublishersContainer.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.wan.impl;
+
+import com.hazelcast.internal.services.ServiceNamespace;
+import com.hazelcast.monitor.LocalWanPublisherStats;
+import com.hazelcast.spi.partition.PartitionReplicationEvent;
+import com.hazelcast.wan.WanReplicationEvent;
+import com.hazelcast.wan.WanReplicationPublisher;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.hazelcast.util.MapUtil.createHashMap;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * Delegating WAN replication publisher implementation. This implementation
+ * is a container for multiple WAN publishers.
+ * When publishing an event on this delegate, all publishers are notified.
+ */
+public final class WanReplicationPublishersContainer {
```

Comment:
OS has `WanReplicationPublishersContainer` and EE has `WanPublisherContainer`, it seemed a confusion point to me. Names are too close in meaning.

Reply:
I don't have any good ideas. Both are internal classes so I wouldn't be too bothered by it but the best I can come up with is this:
- `WanReplicationPublishersContainer` -> `DelegatingWanReplicationScheme`
- `WanPublisherContainer` -> `WanReplicationSchemeContainer`

That's because if you have this config:
```
<wan-replication name="my-wan-cluster">
        <batch-publisher>
            ...
        </batch-publisher>
    </wan-replication>
```
You're actually defining a WAN replication scheme in WAN lingo.

WDYT?

Reply:
your suggested names are better to me.

Reply:
Addressed in a separate commit.